core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java

```diff
@@ -184,9 +186,11 @@ private PageId getPageId(Path path) {
       if (!fileBucket.equals(getFileBucket(fileId))) {
         return null;
       }
-      String fileName = Preconditions.checkNotNull(matcher.group(3));
-      long pageIndex = Long.parseLong(fileName);
-      return new PageId(fileId, pageIndex);
+      String folderName = Preconditions.checkNotNull(matcher.group(3));
+      long pageIndex = Long.parseLong(folderName);
+      String fileName = Preconditions.checkNotNull(matcher.group(4));
+      long mTime = Long.parseLong(fileName);
```

Comment:
is the filename here always a timestamp?

Reply:
> Thank you @frankobe for this PR!
> 
> I'm not sure if there is any other impact if we put an extra field in the pageId. it might be good for us to implement the TTL. @apc999 what do you think?
> If we're just talking about presto + alluxio use case, I might prefer to append the timestamp into the cache identifier on presto side. Just like this PR did. [prestodb/presto#16453](https://github.com/prestodb/presto/pull/16453)

We prefer to separate the modified time out to make it easier to implement TTL. The same reason applies for the presto pr. Including modificationTime in fileId makes it hard to reverse calculate the modifiedTime so we make the cache path like 

`root_path/page_size(ulong)/bucket(uint)/file_id(str)/page_idx(ulong)/mtime(long)` on purpose